packages/frontend/src/components/Card/Card.tsx

```diff
@@ -0,0 +1,44 @@
+import { useState } from "react";
+import {
+  Wrapper,
+  ImageContainer,
+  RecipeName,
+  BottomPanel,
+  RatingWrapper,
+  Rating,
+  StarIcon,
+  HeartIcon,
+  Author,
+} from "../Card/Card.styled";
+import starIcon from "src/assets/images/star.svg";
+import heartIcon from "src/assets/images/heart.svg";
+import { Routes } from "src/enums/routes.enum";
+import { CardProps } from "./Card.types";
+
+export const Card: React.FC<CardProps> = ({ id, photoUrl, title, rating, total, author }) => {
+  const [isFilled, setIsFilled] = useState(false);
+
+  const toggleFill = (e: React.MouseEvent<SVGElement, MouseEvent>) => {
+    e.preventDefault();
+    setIsFilled(!isFilled);
+  };
+  return (
+    <Wrapper to={`${Routes.Recipe}${id}`}>
+      <ImageContainer bgImage={photoUrl}></ImageContainer>
+      <HeartIcon src={heartIcon} $isFilled={isFilled} onClick={(e) => toggleFill(e)} />
+
+      <RecipeName>{title}</RecipeName>
+      <BottomPanel>
+        {rating && total ? (
```

Comment:
!!rating && !!total

Reply:
dlaczego tak?

Reply:
bo rating jest typu number i to co robi  opertor && to zwraca wartość jesli pierwsza jest fałszywa to zwróci pierwsza jeśli jest prawdizwa t ozwróci drugą w momencie jak masz typ number string object to może ci się wkraść po prostu liczba gdzieś w kod to zo robi opertor ! to zmienia dowolny typ w boolean i odrwaca jak dasz !! to najpierw parsujesz rating na typ boolean z typu number i odwrcasz jego wartość a potem odwracasz jąjeszcze raz czyli masz docelową wartość rating z parsowanąna boolean